api.ts

```diff
@@ -1767,9 +1831,12 @@ export type MetadataFormulaResultType = string | number | MetadataFormulaObjectR
  * values of the others. This is dictionary mapping the names of each parameter to its
  * current value.
  */
-export type MetadataFormula<ContextT extends ExecutionContext = ExecutionContext> = BaseFormula<
+export type MetadataFormula<
+  ContextT extends ExecutionContext = ExecutionContext,
+  ResultT extends PackFormulaResult = LegacyDefaultMetadataReturnType,
```

Comment:
This change was causing some downstream changes, as TS is not always able to determine if a return type is compatible with LegacyDeafultMetadataReturnType. Thinking it's still preferred to have this be the default and eliminate a usage of `any`; this just means we have to explicitly say what the return type is for some metadata formulas, the one i've seen most often is getConnectionName